Fix minor issues in TEP descriptions

mrosanes opened the pull request:
Fix minor formatting issues in TEP descriptions, images, broken links and missing links.

Commit:
Fix link to the TEPs workflow image

The image representing the workflow to be followed for the TEPs was
broken. Create a folder of figures, put the workflow image there
and fix the link.

mrosanes committed:
Fix broken links

Fix broken links

mrosanes committed:
Fix markdown formatting 

Fix markdown formatting. 
Change some question marks characters to double quotation marks.

mrosanes committed:
Fix markdown formatting in TEP3

mrosanes committed:
Fix broken links in TEP3

Fix broken links in TEP3

mrosanes committed:
Fix broken link in TEP7

Fix broken link related to Taurus coding conventions in TEP7

mrosanes committed:
Fix formatting problem

Fix formatting problem in the branch names.

mrosanes committed:
Fix formatting

Fix formatting in "Changes" section from TEP8.

mrosanes committed:
Fix formatting in table 

Fix formatting problem in table and fix 
carriage returns.

mrosanes committed:
Fix table formatting

Fix formatting in the second Taurus Logger API table.

mrosanes committed:
Fix minor formatting problems

Fix some formatting in the text of TEP13

mrosanes committed:
Add some links

Add missing links in TEP13

mrosanes committed:
Modify formatting in TEP13 

Modify formatting in TEP13

mrosanes committed:
Add missing links in TEP14

Add missing links in TEP14 description

mrosanes committed:
Fix table in TEP14

Fix table of types in TEP14

Commit:
Rename folder for resources

Rename folder where TEP resources (images, etc.) will be located.

Commit:
Rename tep0 workflow figure

Rename tep0 figure representing the TEP cycle workflow.

mrosanes committed:
Fix link to figure on TEP0

Fix link to figure on TEP0

Commit:
Modify figure location in TEP14

Modify figure location in TEP14 and correct its link.

cpascual commented on doc/source/tep/TEP14.md:
```diff
@@ -109,13 +109,13 @@ Note that this change is one of the possible sources of unavoidable backwards in
 ============================
 
-The concept of "Authority" models type was already introduced in [TEP3] to generalise the tango-centric "Database". This TEP defines a base API for xxxAuthority via the TaurusAuthority base class. This is the api that scheme-agnostic code can use for dealing with Authority models. 
+The concept of "Authority" models type was already introduced in [TEP3](https://sourceforge.net/p/tauruslib/wiki/TEP3/) to generalise the tango-centric "Database". This TEP defines a base API for xxxAuthority via the TaurusAuthority base class. This is the api that scheme-agnostic code can use for dealing with Authority models. 
```
I would prefer to avoid filling the text with links to the same page over and over. I suggest one of these two alternatives:
a) the link is only added the first time the page is mentioned
b) the reference-style syntax is used in all cases:  

```
blablablabla [TEP3][] blanblblablablabla [TEP3][] blabla
(...)
[TEP3]: https://sourceforge.net/p/tauruslib/wiki/TEP3/
```

I personally, prefer the second if the same page is linked several times

mrosanes replied:
Applied second alternative as suggested, when a page is linked several times.

cpascual commented on doc/source/tep/TEP14.md:
```diff
@@ -17,7 +17,7 @@
 =========================
 
-This TEP is a spin-off of [TEP3]. It proposes a refactoring of taurus.core to get rid of some tango-centric implementation details that were left by [TEP3]. The objective is to obtain a cleaner and more scheme-agnostic API for creation of new schemes. Several inter-related aspects of the taurus core are refactored:
+This TEP is a spin-off of [TEP3](https://sourceforge.net/p/tauruslib/wiki/TEP3/). It proposes a refactoring of taurus.core to get rid of some tango-centric implementation details that were left by [TEP3](https://sourceforge.net/p/tauruslib/wiki/TEP3/). The objective is to obtain a cleaner and more scheme-agnostic API for creation of new schemes. Several inter-related aspects of the taurus core ar ...
```
Eventually, all these TEP URLs should be converted to the host-agnostic ones (e.g. `http://www.taurus-scada.org/tep.html?TEP3.md`)

mrosanes replied:
Changes applied in most of the URLs. For the SEPs, and some of the taurus URLs, the URLs are still the old ones pointing to sourceforge. They can be corrected in the near future, once those URLs exists on github or on taurus-scada.org.

cpascual commented on doc/source/tep/TEP3.md:
```diff
@@ -245,13 +245,13 @@ would now be written as:
 -----------------------------------------------------------------------------
 
-The proposed implementation changes the model name syntax for the configuration objects of both the tango and eval schemes. Where before the suffix "?configuration" was used, now "#" is used. Similarly, when a configuration key was defined with the suffix "?configuration=<cfg_key>", now it is defined as "#<cfg_key>". 
+The proposed implementation changes the model name syntax for the configuration objects of both the tango and eval schemes. Where before the suffix "?configuration" was used, now "#" is used. Similarly, when a configuration key was defined with the suffix "?configuration=\<cfg_key\>", now it is defined as "#\<cfg_key\>". 
```
Instead of "?configuration=<cfg_key>" better use "?configuration=`<cfg_key>`"

mrosanes replied:
Corrected.

cpascual commented on doc/source/tep/TEP8.md:
```diff
@@ -31,60 +31,60 @@ Taurus Logger has to be created in two steps: Taurus Logger initialization and T
-taurus/lib/taurus/qt/qtgui/table/qlogtable.py
-taurus/lib/taurus/qt/qtgui/taurusgui/taurusgui.py
-taurus/lib/taurus/qt/qtgui/tree/taurusdevicetree.py
+src/sardana/macroserver/macro.py<br>
```
Instead of using `<br>` html tags, do the same enclosing the full list with triple-backquotes ( ``` )

mrosanes replied:
Corrected.

Commit:
Modify URLs of TEPs documentation

Modify URLs pointing to TEPs documentation.

cpascual reviewed (commented):
All seems generally  ok, but a few changes are needed

Commit:
Minor changes in TEP8 formatting

Minor changes in TEP8 formatting.
Instead of using <br> html tags, do the same enclosing 
the full list with triple-backquotes ( ``` ).

Commit:
Use reference-style link in TEP14

Modify links formatting.
Use reference-style link in TEP14 description.

Commit:
Use reference-style link in TEP3

Modify links formatting.
Use reference-style links in TEP3 description
if pages are linked several times.

Commit:
Use reference-style link in TEP7

Modify links formatting.
Use reference-style links in TEP7 description
if pages are linked several times.

Commit:
Use reference-style link in TEP13

Modify links formatting.
Use reference-style links in TEP13 description
if pages are linked several times.

Commit:
Fix minor formatting in TEP3

Fix minor formatting in TEP3.

Commit:
Modify URLs in TEP0

Modify TEP URLs. 
Eventually, all TEP URLs should be converted to the host-agnostic ones 
(e.g. http://www.taurus-scada.org/tep.html?TEP0.md)

Commit:
Modify URLs in TEP3

Modify TEP URLs. 
Eventually, all TEP URLs should be converted to the host-agnostic ones 
(e.g. http://www.taurus-scada.org/tep.html?TEP0.md)

Commit:
Modify URL in TEP7

Modify TEP URL. 
Eventually, all TEP URLs should be converted to the host-agnostic ones 
(e.g. http://www.taurus-scada.org/tep?TEP0.md)

Commit:
Modify URL in TEP8

Modify TEP URL. 
Eventually, all TEP URLs should be converted to the host-agnostic ones 
(e.g. http://www.taurus-scada.org/tep?TEP0.md)

Commit:
Modify URLs in TEP13

Modify TEP URLs. 
Eventually, all TEP URLs should be converted to the host-agnostic ones 
(e.g. http://www.taurus-scada.org/tep?TEP0.md)

Commit:
Modify URLs in TEP14

Modify TEP URLs. 
Eventually, all TEP URLs should be converted to the host-agnostic ones 
(e.g. http://www.taurus-scada.org/tep?TEP0.md)

Commit:
Modify URL in TEP15

Modify TEP URL. 
Eventually, all TEP URLs should be converted to the host-agnostic ones 
(e.g. http://www.taurus-scada.org/tep?TEP0.md)

Commit:
Modify URL in TEP16

Modify TEP URL. 
Eventually, all TEP URLs should be converted to the host-agnostic ones
(e.g. http://www.taurus-scada.org/tep?TEP0.md)

cpascual commented on doc/source/tep/TEP0.md:
```diff
@@ -199,7 +199,7 @@ start with a paragraph of the following RFC822-style headers:
   syntax for the To: header
 * URL: during DRAFT state, a link to the wiki place of the draft
-  (typically probably https://sourceforge.net/p/tauruslib/wiki/TEPxxx)
+  (typically probably http://www.taurus-scada.org/tep/?TEPxxx.md)
```
If we mark the TEP0 as obsolete, maybe we shouldn't change the URLs referring to wikis (other than the header).

what do you think?

cpascual commented on doc/source/tep/TEP0.md:
```diff
@@ -3,7 +3,7 @@
     State: ACCEPTED
```
We should change this to OBSOLETE and add a reason to the header (pointing to TEP16).

This applies to all TEPS obsoleted by TEP16 (see TEP16 text for a list)

cpascual commented on doc/source/tep/TEP14.md:
```diff
@@ -225,3 +225,6 @@ Changes
 
 2016-03-16
 [cpascual](http://sf.net/u/cpascual/) Passing to  ACCEPTED
+
```
A change entry should be added in **all** the TEPs indicating that URLs and format were updated according with TEP16

Commit:
Change TEP0 state to obsolete

Change state of TEP0 to obsolete according to TEP16

Commit:
Change TEP7 state to obsolete

TEP16 obsoletes TEP7.
Change TEP7 state to obsolete.

Commit:
Modify links to point to wiki in TEP0

Modify links to point to sourceforge wiki.
TEP0 becomes obsolete. Links must be 
kept pointing to sourceforge wiki.

Commit:
Modify history of changes

Add entry to change history

Commit:
Add change entry in change history

Add change entry in change history on all TEPs, indicating the
adaptation of TEPs introduced by TEP16.

Commit:
Remove entry in change history

Remove incorrect entry in change history

Commit:
Change Reason header in TEP0

Commit:
Update TEP0.md

Commit:
(minor) remove colon (for format coherence)

Commit:
Remove SF-specific wiki formatting

Commit:
Modify Reason header

Commit:
Fix changes section

Commit:
Fix Changes section
